### DISCUSSION TITLE
data preprocessing

### DISCUSSION THREAD

**Discussion by oralb** [cropbox/Cropbox.jl#13] [Q&A] [answered]
Cropbox seems to require a specific data format (i.e., units in column names; hyphens as a separator for date columns ). Although I can figure out this using examples or error messages, documentation would also be helpful. Where can I find the description to trim the data before importing data into Cropbox?

**Comment by tomyun** [comment, marked as answer]
As you mentioned, `unitfy()` function can take a DataFrame and assign measurement units or certain data types to its columns. When the name of column is in the form of `"name_of_column (unit_or_type)"`, the `unit_or_type` part becomes subject to automatic unit/type assignment.

1. Here are some examples of handling units.

- `"leaf_area (cm)"` becomes `"leaf_area"` column with `u"cm"` unit, i.e. `Quantity{Float64, ...}`.
- `"A_net (μmol/m^2/s)"` becomes `"A_net"` column with `u"μmol/m^2/s"` unit.

2. In the case of type assignment, currently we only support one keyword `:date`.

- `"observation (:date)"` becomes `"observation"` column where date-formatted strings (e.g. `"2022-11-26"`) are explicitly converted to `Date` type (e.g. `Date(2022, 11, 26)`) from Julia standard `Dates` module.
- FYI, no support for `:datetime` at the moment due to the complexity of time zone handling.

You may get a better idea how they are actually handled by looking at [the code](https://github.com/cropbox/Cropbox.jl/blob/d1b40c11bce951f1a85355748a21d193bf6873a6/src/unit.jl#L65-L87).
